app/src/androidTest/java/org/mozilla/reference/browser/ui/robots/AddToHomeScreenRobot.kt

```diff
@@ -16,11 +19,22 @@ class AddToHomeScreenRobot {
         cancelAddToHomeScreenButton().click()
     }
 
-    fun clickAddAutomaticallyToHomeScreenButton() {
-        addAutomaticallyToHomeScreenButton().waitForExists(waitingTime)
+    fun clickSystemHomeScreenShortcutAddButton() {
+        when (Build.VERSION.SDK_INT) {
+            in Build.VERSION_CODES.O..Build.VERSION_CODES.R -> clickAddAutomaticallyButton()
+            in Build.VERSION_CODES.S..Build.VERSION_CODES.UPSIDE_DOWN_CAKE -> clickAddToHomeScreenButton()
```

Comment:
I would just add a comment stating that this is up to Android 14 as a known tested version